pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -516,11 +511,17 @@ func buildInboundCatchAllHTTPFilterChains(configgen *ConfigGeneratorImpl,
 			}
 		}
 
+		listenerOpts := buildListenerOpts{
+			push:  push,
+			proxy: node,
+			bind:  matchingIP,
+			port:  port,
+		}
 		// Construct the actual filter chains for each of the filter chain from the plugin.
 		for _, chain := range allChains {
-			httpOpts := configgen.buildSidecarInboundHTTPListenerOptsForPortOrUDS(node, in)
+			httpOpts := configgen.buildSidecarInboundHTTPListenerOptsForPortOrUDS(node, in, clusterName)
```

Comment:
it is not worthwhile to remove the InboundClusterName from plugin input params and just pass it as an additional argument - all call sites will have a mystery "" argument in addition to plugin input params. Can we please restore it?

Reply:
I would say the opposite - it does not make sense to use 'plugin params' in the main code.

The problem is we have this huge struct for "plugin options" which at any given point has only some subset of fields set. Then all over the code we use random fields there or in listener lots or as plain parameters. Sometimes a given function has two "port" fields set which are entirely different. It makes the entire code extremely hard to reason about